samza-sql-lib/src/main/java/org/apache/samza/sql/data/SamzaSqlRelMessage.java

```diff
@@ -0,0 +1,52 @@
+/*
+* Licensed to the Apache Software Foundation (ASF) under one
+* or more contributor license agreements.  See the NOTICE file
+* distributed with this work for additional information
+* regarding copyright ownership.  The ASF licenses this file
+* to you under the Apache License, Version 2.0 (the
+* "License"); you may not use this file except in compliance
+* with the License.  You may obtain a copy of the License at
+*
+*   http://www.apache.org/licenses/LICENSE-2.0
+*
+* Unless required by applicable law or agreed to in writing,
+* software distributed under the License is distributed on an
+* "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+* KIND, either express or implied.  See the License for the
+* specific language governing permissions and limitations
+* under the License.
+*/
+
+package org.apache.samza.sql.data;
+
+import java.util.Optional;
+
+
+public class SamzaSqlRelMessage {
+
+  private final Object[] value;
+  private final String[] names;
+
+  public SamzaSqlRelMessage(Object[] values, String[] names) {
+    this.value = values;
+    this.names = names;
```

Comment:
It's generally a good idea to prefer value objects where possible. It looks like that is the intention with this class (though the value cells may be mutable depending on the implementation). However, you can provide a stronger guarantee if you use List instead of Array and guard the value and names lists with an immutable wrapper. Otherwise it is possible to change the either through the returned arrays (getNames / getValue).